packages/integrations/mdx/src/rehype-collect-headings.ts

```diff
@@ -0,0 +1,49 @@
+import Slugger from 'github-slugger';
+import { visit } from 'unist-util-visit';
+import type { VFile } from 'vfile';
+
+export interface MarkdownHeading {
+	depth: number;
+	slug: string;
+	text: string;
+}
+
+const slugger = new Slugger();
```

Comment:
Hoisting this up here might dedupe headings _across_ pages? I think you need to do `slugger.reset()` (I can't remember the exact API) each time this runs or just move this constructor inside the plugin.

Reply:
Oh, didn't know that! I'll move slugger inside the runner 👍